crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1270,16 +1270,13 @@ impl PgIndexerStore {
     ) -> Result<(), IndexerError> {
         let mutated_objects: Vec<Object> = tx_object_changes
             .iter()
-            .flat_map(|changes| changes.changed_objects.iter().cloned())
-            .collect();
-        let deleted_changes = tx_object_changes
-            .iter()
-            .flat_map(|changes| changes.deleted_objects.iter().cloned())
-            .collect::<Vec<_>>();
-        let deleted_objects: Vec<Object> = deleted_changes
-            .iter()
-            .map(|deleted_object| deleted_object.clone().into())
+            .flat_map(|changes| changes.changed_objects.iter())
+            .map(|changed_object| (changed_object.object_id.as_str(), changed_object))
+            .collect::<HashMap<_, _>>()
```

Comment:
Just to verify: is `collect::<HashMap>()` going to pick the latest value in case there are multiple objects with the same id?

Reply:
yes, it'll pickup the later value. These lists of changes *should* be ordered by transaction order so that would be the expected/desired result